diff --git a/glamor/glamor_egl.h b/glamor/glamor_egl.h
index 2f7566b24..8f6ed7840 100644
--- a/glamor/glamor_egl.h
+++ b/glamor/glamor_egl.h
@@ -28,6 +28,7 @@
 #define GLAMOR_EGL_H
 
 #define MESA_EGL_NO_X11_HEADERS
+#define EGL_NO_X11
 #include <epoxy/gl.h>
 #include <epoxy/egl.h>
 #include <glamor_egl_ext.h>
diff --git a/glamor/glamor_priv.h b/glamor/glamor_priv.h
index 10c74c1b5..ea7a8bc96 100644
--- a/glamor/glamor_priv.h
+++ b/glamor/glamor_priv.h
@@ -40,6 +40,7 @@
 #include <epoxy/gl.h>
 #ifdef GLAMOR_HAS_GBM
 #define MESA_EGL_NO_X11_HEADERS
+#define EGL_NO_X11
 #include <epoxy/egl.h>
 #endif
 
diff --git a/glx/meson.build b/glx/meson.build
index 7f9e549f0..1140522f6 100644
--- a/glx/meson.build
+++ b/glx/meson.build
@@ -39,7 +39,7 @@ if build_glx
             common_dep,
             dl_dep,
             dependency('glproto', version: '>= 1.4.17'),
-            dependency('gl', version: '>= 9.2.0'),
+            dependency('gl', version: '>= 1.2'),
         ],
         c_args: [
             glx_align64,
@@ -74,7 +74,7 @@ if build_glx
             common_dep,
             dl_dep,
             dependency('glproto', version: '>= 1.4.17'),
-            dependency('gl', version: '>= 9.2.0'),
+            dependency('gl', version: '>= 1.2'),
         ],
     )
 
diff --git a/hw/xwayland/xwayland-glamor-eglstream.c b/hw/xwayland/xwayland-glamor-eglstream.c
index c62c0d2ac..36b749aaf 100644
--- a/hw/xwayland/xwayland-glamor-eglstream.c
+++ b/hw/xwayland/xwayland-glamor-eglstream.c
@@ -33,6 +33,7 @@
 #include "wayland-eglstream-controller-client-protocol.h"
 
 #define MESA_EGL_NO_X11_HEADERS
+#define EGL_NO_X11
 #include <glamor_egl.h>
 #include <glamor.h>
 #include <glamor_transform.h>
diff --git a/hw/xwayland/xwayland-glamor-gbm.c b/hw/xwayland/xwayland-glamor-gbm.c
index c13515c3c..7ccbc647e 100644
--- a/hw/xwayland/xwayland-glamor-gbm.c
+++ b/hw/xwayland/xwayland-glamor-gbm.c
@@ -36,6 +36,7 @@
 #include <drm_fourcc.h>
 
 #define MESA_EGL_NO_X11_HEADERS
+#define EGL_NO_X11
 #include <gbm.h>
 #include <glamor_egl.h>
 
diff --git a/hw/xwayland/xwayland-glamor.c b/hw/xwayland/xwayland-glamor.c
index dc07f7933..bf7f0ffb4 100644
--- a/hw/xwayland/xwayland-glamor.c
+++ b/hw/xwayland/xwayland-glamor.c
@@ -26,6 +26,7 @@
 #include "xwayland.h"
 
 #define MESA_EGL_NO_X11_HEADERS
+#define EGL_NO_X11
 #include <glamor_egl.h>
 
 #include <glamor.h>
diff --git a/hw/xwayland/xwayland-glx.c b/hw/xwayland/xwayland-glx.c
index abf48f71c..b0b227d87 100644
--- a/hw/xwayland/xwayland-glx.c
+++ b/hw/xwayland/xwayland-glx.c
@@ -33,6 +33,7 @@
 #include <xwayland-config.h>
 #include "xwayland.h"
 #define MESA_EGL_NO_X11_HEADERS
+#define EGL_NO_X11
 // #include <EGL/egl.h>
 #include <epoxy/egl.h>
 #include "glxserver.h"
